ios/engine/KMEI/KeymanEngine/Classes/KeymanPackage.swift

```diff
@@ -54,6 +54,107 @@ public class KeymanPackage {
     }
   }
 
+  /**
+   * Indicates the distribution level and state of the package.
+   */
+  public enum SupportState: String, Codable {
+    /**
+     * Indicates that the support level for the package is unknown.  Generally occurs when a
+     * package is installed via file-sharing before KeymanEngine is able to perform a package-version check.
+     */
+    case unknown
+
+    /**
+     * Indicates that the package is publicly distributed but no longer maintained.  Indicates that other packages targetting
+     * the same targets are more favored.
+     */
+    case deprecated
+
+    /**
+     * Indicates that this package receives full support and maintenance from the resource development community at large.
+     */
+    case publiclyReleased = "publicly released"
```

Comment:
I'm not very comfortable with this designation and its comment (and thus also `deprecated` and `custom`). It implies a little more knowledge than we can reasonably assume from the package-version query. I think instead, we need to view this a little less in terms of "support state" and more in terms of "mode of distribution":

* `unknown`: we don't know if it's on cloud or not
* `cloud-deprecated`: distributed via Keyman Cloud, but deprecated
* `cloud`: distributed via Keyman Cloud and not deprecated (but we cannot infer support level etc from that)
* `custom`: not distributed via Keyman Cloud (aka ad-hoc, peer-to-peer). Doesn't infer whether or not it is publicly distributed as we cannot determine that.

This suggests that `SupportState` is probably the wrong name for the enum. Maybe instead `DistributionState`?